mautic/mautic
# TPROD-346 Transifex SDK update

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [n]
| Deprecations?                          | [n]
| BC breaks? (use the c.x branch)        | [y]
| Automated tests included?              | [ y 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-346



#### Description:

This PR implements new Transifex SDK version:

https://github.com/mautic/Transifex-API/pull/4

And improves the code related to pushing and pulling translations.

The API version we use today will stop work at the end of the month!

The old `mautic:transifex:push` used to only update resources. It never created the new ones. It had a `--create` option for that. The new command checks whether the resource exists and decides whether it needs to create or update it. So it may happen that once we run the new command for the first time we'll get a bunch of new resources created.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. You must have admin access to the mautic/mautic Transifex project, so skip testing this if you don't.
3. Generate new API key: https://developers.transifex.com/reference/api-authentication
4. Store it in `app/config/local.php` under key `transifex_api_token`
5. Test the `bin/console mautic:transifex:pull` command. It should download all translations that are translated from at least 80%. This constraint is checked for every file. Not by language. Check that files in the `translations` directory are being updated or created based on if the file existed before or not. Feel free to use the `--language` and `--bundle` filters to test for only certain languages or bundles.
6.  `bin/console mautic:transifex:push` command. ⚠️  **this is going to push to the production Transifex project!** so possibly use the `--bundle` option to push only files for one bundle in case this I made a mistake somewhere. Then check in the Transifex UI that the resource is still translatable.
